src/java.desktop/unix/classes/sun/awt/wl/WLFrameDecoration.java

```diff
@@ -352,6 +365,15 @@ void setActive(boolean active) {
         }
     }
 
+    void setHasServerSideDecoration(boolean hasServerSideDecoration) {
+        boolean didNotDrawClientSideDecoration = doesNotDrawClientSideDecoration();
+        this.hasServerSideDecoration = hasServerSideDecoration;
+        if (doesNotDrawClientSideDecoration() != didNotDrawClientSideDecoration) {
```

Comment:
I'm not sure this entire if is needed. Can you explain what this code is supposed to be doing?

Reply:
> When the sever changes the decoration mode between server-side and client-side, the decorations must be re-drawn.

Can this happen on the fly? Have you tested such scenario?

Reply:
It can in some compositors. Either manually or when the window switches between tiling and floating. But I realized that I had forgotten to generated a layout event in that case. This should be fixed in the latest commit.

Reply:
>Have you tested such scenario?

I have now.